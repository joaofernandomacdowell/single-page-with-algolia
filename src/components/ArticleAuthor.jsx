import React from 'react';

const ArticleAuthor = ({ author }) =>
  <p className="zoom-article-author">{author.name}</p>

export default ArticleAuthor;