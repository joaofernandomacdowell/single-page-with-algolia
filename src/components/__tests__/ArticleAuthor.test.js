import React from 'react'
import { shallow } from 'enzyme';

import mockApiResponse from '../../helper';
import ArticleAuthor from '../ArticleAuthor';


describe('ArticleAuthor component', () => {
  const props = {
    author: mockApiResponse.hits[0].author
  };
  const wrapper = shallow(<ArticleAuthor {...props} />);

  test('match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('have author label', () => {
    const label = wrapper.find('.zoom-article-author');

    expect(label).toHaveLength(1);
    expect(label.text()).toBe(mockApiResponse.hits[0].author.name);
  });
});