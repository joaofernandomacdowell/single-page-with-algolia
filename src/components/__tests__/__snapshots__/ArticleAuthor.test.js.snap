// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleAuthor component match the snapshot 1`] = `ShallowWrapper {}`;
